### ISSUE TITLE
[Feature Request]: swaync  support

### ISSUE THREAD

**Issue by kRHYME7** [HyDE-Project/HyDE#39]
swaync wallbash and selector script?

**Comment by imtheabdi** [comment]
> swaync wallbash and selector script?

would love a notification center for sure! a material theme style one would look really sweet!! ;p

**Comment by brunodunbar** [comment]
I had installed it manually, but I suck at design and theming, it would be lovely to have it properly integrated with the theming system.

**Comment by dieBakterie** [comment]
> swaync wallbash and selector script?

I think i've those in my repo it's not really good but it's an starting point, it themes with the system(has wallbash support) **but** it's originally for **Hyprdots/old HyDE**...

**Comment by kRHYME7** [comment]
Hi @dieBakterie feel free to contribute. Let's discuss in there what can be added.

**Comment by dieBakterie** [comment]
Hi @kRHYME7 we could implement in themeswitscher to change the source line to the according theme css file for swaync and with wallbash we change the line to the wallbash css and write the colours in ther. I think that's how it's done right now

**Comment by dieBakterie** [comment]
Therefore we need the selection between dunst and swaync for example since only one of them should be installed.

**Comment by kRHYME7** [comment]
> Therefore we need the selection between dunst and swaync for example since only one of them should be installed.


Nah, doesn't matter. Whichever is running first prevails. Also we have `$start.NOTIFICATION` so setting this to swaync will work.

**Comment by dieBakterie** [comment]
Okay should we've an own css file fo each theme and symlink it to the swanc folder or should we've the theme specific swanc files in the swanc folder instead of the themes? We could just launch swaync complicated via script and set the theme this way, example `exec-once swaync --config ~/.config/swaync/config.jsonc --theme $themepath` this way we coulkd use a jsonc and have comments since just `exec-once swaync` only works with config.json being in `~/.config/swaync`. We could just change the import/source line to match the current theme specific css and for wallbash we just do it like for waybar for example, just need to unify some swaync stuff, change "var names" to match the default ones.

Two variants of doing it rn that i've implemented are one wallbash or two have an theme specific css and update the source/import line in the config file via sed, alternative version only needs this and has a theme specific file for each existing theme as of hyprdots july 2024 i would say, needs to be updated to match the themes and added for new ones ofcourse

```
# swaync
#! only for alternative version needed
if pkg_installed swaync || pkg_installed swaynotificationcenter; then
     style_css="${confDir}/swaync/style.css"

     # convert themeSet to the format used in style.css
     theme_file=$(echo "$gtkTheme" | tr '[:upper:]' '[:lower:]' | tr '-' '_') # easiest way of getting all hyde theme names

     # replaces the @import-Statement with the desired theme
     sed -i "s|@import \"\.\/themes\/[^\"]*\.css\"|@import \"\.\/themes\/${theme_file}.css\"|g" "${style_css}"
 fi
```
other version don't needs this, well it's just wallbash but i think we should use both the alternative vesion and then wallbash when wallbash is activated/used so just like now in general.
[wallbashswaync.sh](https://github.com/dieBakterie/HyDE-Clone/blob/477facfbcb20c0b42491819f596501c8e66a461d/Configs/.local/share/bin/wallbashswaync.sh)

**Comment by dieBakterie** [comment]
> +1 for the need of a notification center. Maybe a good starting point is https://github.com/mylinuxforwork/dotfiles/blob/main/share/dotfiles/.config/waybar/modules.json#L170 ?

That's stock swaync waybar module from wiki.

**Comment by kRHYME7** [comment]
TBH this dots I got for swaync is almost a year now lol. But the swaync author is willing to make it better ( multi file). That should make it easier. 


<img width="3000" height="2000" alt="Image" src="https://github.com/user-attachments/assets/00164b5e-d0dc-473b-9f50-c7173c0fe43f" />

<img width="3000" height="2000" alt="Image" src="https://github.com/user-attachments/assets/2464362a-e4cf-4e48-81a1-d1b1471c19b4" />


---

for anyone wondering we already have this in HyDE but it will not automatically be populated as we need to fine tune some stuff. 

<img width="962" height="186" alt="Image" src="https://github.com/user-attachments/assets/0f4367d5-3f95-445e-8510-a7f3f162d0c9" />

**Comment by kRHYME7** [comment]
Closing this as swaync is in the repo but still  experimental.
